src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -208,7 +214,7 @@ private BoundExpression MakeBinaryOperator(
                     case BinaryOperatorKind.ObjectAndStringConcatenation:
                     case BinaryOperatorKind.StringAndObjectConcatenation:
                     case BinaryOperatorKind.StringConcatenation:
-                        return RewriteStringConcatenation(syntax, operatorKind, loweredLeft, loweredRight, type);
+                        throw ExceptionUtilities.UnexpectedValue(operatorKind);
```

Comment:
>throw ExceptionUtilities.UnexpectedValue(operatorKind); [](http://example.com/codeflow?start=24&length=55)

It looks like ```MakeBinaryOperator``` helper is used in many places. Would it make sense to adjust the code to handle this code path as a stand-alone operator?

Reply:
It could be, but I don't think we need to do so right now. The code that I've written presumes that the input has not yet been visited, and `MakeBinaryOperator` is only called on already-visited operators. While we could update the new code to accept both pre-visited and post-visited operations, I think we can save that exercise for when we encounter a case that needs them.